### PR TITLE
fix: spark may not show exceptions or show backtrace as json

### PR DESCRIPTION
**Pull request by kenjis**
**Description**

Fix a bug that `spark` may not show exceptions in production mode, or show backtrace as json.

**How to Reproduce**

(1) Setup MySQL and use the default Config (no `.env`). That is `database` is an empty string and in `production` mode.

Before:
```console
$ php spark db:table

CodeIgniter v4.4.1 Command Line Tool - Server Time: 2023-09-25 02:24:59 UTC+00:00
```

After:
```console
$ php spark db:table

CodeIgniter v4.4.1 Command Line Tool - Server Time: 2023-09-25 02:23:08 UTC+00:00

[mysqli_sql_exception]

You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1

at SYSTEMPATH/Database/MySQLi/Connection.php:306
```

(2) Change to `development` mode.

Before:
```console
$ php spark db:table

CodeIgniter v4.4.1 Command Line Tool - Server Time: 2023-09-25 01:52:27 UTC+00:00

{
    "title": "mysqli_sql_exception",
    "type": "mysqli_sql_exception",
    "code": 500,
    "message": "You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1",
    "file": "/Users/kenji/work/codeigniter/official/CodeIgniter4/system/Database/MySQLi/Connection.php",
    "line": 306,
    "trace": [
        {
...
```

After:
```console
$ php spark db:table

CodeIgniter v4.4.1 Command Line Tool - Server Time: 2023-09-25 02:36:43 UTC+00:00


[mysqli_sql_exception]

You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '' at line 1

at SYSTEMPATH/Database/MySQLi/Connection.php:306

Backtrace:
  1    SYSTEMPATH/Database/MySQLi/Connection.php:306
       mysqli()->query('SHOW TABLES FROM ', 0)
...
```

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Comment by kenjis** [comment]
Added the docs.

File: system/CLI/Console.php [hunk 1]
```diff
@@ -31,6 +32,10 @@ class Console
      */
     public function run()
     {
+        // Create CLIRequest
+        $appConfig = config(App::class);
```
Comment (datamweb) [inline system/CLI/Console.php:36]:
```suggestion
        $appConfig = config('App');
```

Comment (neznaika0) [reply system/CLI/Console.php:36]: Why? `App::class` better

Comment (datamweb) [reply system/CLI/Console.php:36]: You're right, I forgot I was reviewing CI PR.

**Comment by kenjis** [comment]
Rebased to resolve a conflict.

**Review by MGatner — APPROVED** [review]

**Pull request merged**